plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestingOpenLdapServer.java

```diff
@@ -0,0 +1,175 @@
+import static io.trino.plugin.password.ldap.LdapUtil.buildLdapGroupObject;
+import static io.trino.plugin.password.ldap.LdapUtil.buildLdapOrganizationObject;
+import static io.trino.plugin.password.ldap.LdapUtil.buildLdapUserObject;
+import static io.trino.plugin.password.ldap.LdapUtil.randomSuffix;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TestingOpenLdapServer
+        implements Closeable
+{
+    private static final String BASE_DISTINGUISED_NAME = "dc=trino,dc=testldap,dc=com";
+
+    private static final int LDAP_PORT = 389;
+
+    private final Closer closer = Closer.create();
+    private final GenericContainer<?> openLdapServer;
+
+    public TestingOpenLdapServer()
+    {
+        openLdapServer = new GenericContainer<>("ghcr.io/trinodb/testing/centos7-oj11-openldap:" + TestingProperties.getDockerImagesVersion())
+                .withExposedPorts(LDAP_PORT)
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .waitingFor(new HostPortWaitStrategy())
+                .withStartupTimeout(Duration.ofMinutes(5));
+
+        closer.register(openLdapServer::close);
+    }
+
+    public void start()
+    {
+        openLdapServer.start();
+    }
+
+    public String getLdapUrl()
+    {
+        return format("ldap://%s:%s", openLdapServer.getContainerIpAddress(), openLdapServer.getMappedPort(LDAP_PORT));
+    }
+
+    public DisposableSubContext createOrganization()
+            throws NamingException
+    {
+        DirContext context = createContext();
+        try {
+            return new DisposableSubContext(addLdapDefinition(buildLdapOrganizationObject("organization_" + randomSuffix(), BASE_DISTINGUISED_NAME), context));
+        }
+        finally {
+            context.close();
+        }
+    }
+
+    public DisposableSubContext createGroup(DisposableSubContext organization)
+            throws Exception
+    {
+        DirContext context = createContext();
+        try {
+            return new DisposableSubContext(addLdapDefinition(buildLdapGroupObject(organization.getDistinguishedName(), "group_" + randomSuffix()), context));
+        }
+        finally {
+            context.close();
+        }
+    }
+
+    public DisposableSubContext createUser(DisposableSubContext organization, String userName, String password)
+            throws Exception
+    {
+        DirContext context = createContext();
+        try {
+            return new DisposableSubContext(addLdapDefinition(buildLdapUserObject(organization.getDistinguishedName(), userName, password), context));
+        }
+        finally {
+            context.close();
+        }
+    }
+
+    public void addUserToGroup(DisposableSubContext userDistinguishedName, DisposableSubContext groupDistinguishedName)
+            throws Exception
+    {
+        DirContext context = createContext();
+        try {
+            addAttributesToExistingLdapObjects(
```

Comment:
I'm not 100% sure but `memberOf` attribute is a shortcut not available on all LDAP servers.
Shouldn't we also update Group it self ? When memberOf is not available there is group search performed for group contains user in group `member` attribute.

Reply:
This is for ldap server from our image - which supports `memberOf` attribute , in case of other LDAP servers we can have a custom implementation or make it overridden. 

> Shouldn't we also update Group it self

This is only if we don't support `memberOf` right ?

Reply:
Yes, but i believe this attribute can be always created/added manually as it happens in test. 
But this is mostly from AD world where it's being calculated automatically when user is added to group.
So normally you add user to group. And in case of AD all involved users are updated automatically on server side with `memebrOf` attribute.